jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/J9InternalCheckpointHookAPI.java

```diff
@@ -59,7 +59,7 @@ static void registerPostRestoreHook(int priority, String name, Runnable hook) {
 	 * @param hook     the runnable hook
 	 */
 	static void registerPreCheckpointHook(int priority, String name, Runnable hook) {
-		preCheckpointHooks.add(new J9InternalCheckpointHook(priority, name, hook));
+		preCheckpointHooks.add(new J9InternalCheckpointHook(-priority, name, hook));
```

Comment:
why `-priority`?

Reply:
This is to reverse the order of hooks between checkpoint and restore: Eg.

Low Priority
Med Priority
High Priority
------------
checkpoint/Restore
------------
High priority
Med priority
Low priority

Reply:
Can you create the PriorityQueue and pass `Collections.reverseOrder()` to its constructor to get the opposite order?  It would be clearer than this -1.

Reply:
Reading the javadoc for PriorityQueue, I think the `register*` methods need to be synchronized as well